app/controllers/products_controller.rb

```diff
@@ -0,0 +1,152 @@
+class ProductsController < ApplicationController
+  before_action :require_login, except: [:index, :show]
+  before_action :check_product_active_status, only: [:show]
+
+  def index
+    # a guest or merchant can view all products with their details
+    # @products = Product.all
+    @products = Product.where(active: true)
+
+    @order_item = current_order.order_items.new
+    # a guest or merchant can view all products by category
+  end
+
+  def show
+    # product detail page
+    # some conditional visuals based on if you're logged in or not (view-things)
+      # merchants can't buy their own products (i.e. no 'add to cart' button for them)
+      # merchants can't leave reviews for their own products
+      # if instock? == false, you can't add it to the cart
+    @product = Product.find(params[:id])
+    @order_item = current_order.order_items.new
+    @review = Review.new
+
+    @reviews = @product.reviews
+
+    @average = average_rating(@reviews)
+
+    existing_cart_product = current_order.order_items.select {|item| item.product_id == @product.id }
+
+    if existing_cart_product.length == 0
+      @additional_stock_available = @product.stock.to_i
+    elsif existing_cart_product.length > 0
```

Comment:
Seems like this logic could just be in an `else` and the `elsif` logic is extraneous